src/features/search/redux/middleware.ts

```diff
@@ -0,0 +1,21 @@
+import { Middleware, AnyAction } from 'redux'
+import { UPDATE_TICKER } from '../redux'
+import { resetState } from 'redux/actions'
+
+export const searchMiddleware = (socket: SocketIOClient.Socket): Middleware => {
```

Comment:
q: It might be nicer to specify the `SocketIOClient` as a type which you export globally (seeing as all your Middlewares use this type)

Reply:
As opposed to doing this I've implemented Kyle's middleware dependencies type recommendation which effectively carries out what you recommended here.